sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -387,39 +389,48 @@ public virtual async Task RelinquishOwnershipAsync(CancellationToken cancellatio
                     }
                 }
 
-                // Here's the important part.  If there are no processors that have exceeded the maximum owned partition count allowed, we may
-                // need to steal from the processors that have exactly the maximum amount.  If this instance is below the minimum count, then
-                // we have no choice as we need to enforce balancing.  Otherwise, leave it as it is because the distribution wouldn't change.
+                // If this processor has less than the minimum or any other processor has more than the maximum, then we need to steal a partition.
 
-                if (partitionsOwnedByProcessorWithGreaterThanMaximumOwnedPartitionsCount.Count > 0)
+                if ((ownedPartitionsCount < minimumOwnedPartitionsCount) || (partitionsOwnedByProcessorWithGreaterThanMaximumOwnedPartitionsCount.Count > 0))
                 {
-                    // If any stealable partitions were found, randomly pick one of them to claim.
+                    Logger.ShouldStealPartition(OwnerIdentifier);
 
-                    Logger.StealPartition(OwnerIdentifier);
+                    // Prefer stealing from a processor that owns more than the maximum number of partitions.
 
-                    var index = RandomNumberGenerator.Value.Next(partitionsOwnedByProcessorWithGreaterThanMaximumOwnedPartitionsCount.Count);
+                    if (partitionsOwnedByProcessorWithGreaterThanMaximumOwnedPartitionsCount.Count > 0)
```

Comment:
My god, this variable name. Or is it a variable sentence now?! :D

Reply:
I take no credit, but nobody can claim they don't understand its purpose!